frontend/src/individual_add.tsx

```diff
@@ -158,6 +159,106 @@ export function IndividualAdd({
     };
   });
 
+  // creates a breeding event, updates it with birth information and calls the createIndividual function
+  const prepareIndividual = async () => {
+    let breedingString = "";
+    if (individual.birth_date) {
+      let breedingDate: Date | number = new Date(individual.birth_date);
+      breedingDate.setDate(breedingDate.getDate() - 30);
+      breedingString = breedingDate.toLocaleDateString(locale);
+
+      let breeding = {
+        mother: individual.mother?.number,
+        father: individual.father?.number,
+        date: breedingString,
+      };
+
+      let birth = {
+        date: individual.birth_date,
+        litter: individual.litter,
+        id: null,
+      };
+
+      const breedingEvent = await post("/api/breeding", breeding);
+
+      if (breedingEvent.status == "success") {
+        birth = { ...birth, id: breedingEvent.breeding_id };
+      } else if (
+        breedingEvent.status == "error" &&
+        !!breedingEvent.breeding_id // not implemented in backend yet!!
+      ) {
+        birth = { ...birth, id: breedingEvent.breeding_id };
+      } else if (
+        breedingEvent.status == "error" &&
+        breedingEvent.message == "Not logged in"
+      ) {
+        userMessage(
+          "Du är inte inloggad. Logga in och försök igen.",
+          "warning"
+        );
+      } else if (
+        breedingEvent.status == "error" &&
+        breedingEvent.message.includes("Unknown father" || "Unknown mother")
+      ) {
+        userMessage(
+          "En eller båda föräldrar kunde inte hittas. Både mor och far måste vara aktiva individer i databasen.",
+          "warning"
+        );
+      } else if (
+        breedingEvent.status == "error" &&
+        breedingEvent.massage == "Forbidden"
```

Comment:
```suggestion
        breedingEvent.message == "Forbidden"
```

Reply:
But I'm sorry if that was unclear, I don't have a problem with generic messages. And if we want to offer better messages in the frontend, I think we should define error codes rather than rely on strings like htis.

Reply:
Yes, I agree! I'd also love to have some error codes instead of strings. It would be easier and also more in line with standards. Now even an error from the backend becomes a "200 OK" in the browser.